packages/abstractions/src/requestInformation.ts

```diff
@@ -108,20 +111,30 @@ export class RequestInformation {
     contentType?: string | undefined,
     value?: T[] | T
   ): void => {
-    const writer = this.getSerializationWriter(
-      requestAdapter,
-      contentType,
-      value
-    );
-    if (!this.headers) {
-      this.headers = {};
-    }
-    if (Array.isArray(value)) {
-      writer.writeCollectionOfObjectValues(undefined, value);
-    } else {
-      writer.writeObjectValue(undefined, value);
-    }
-    this.setContentAndContentType(writer, contentType);
+    trace
+      .getTracer(RequestInformation.tracerKey)
+      .startActiveSpan("setContentFromParsable", (span) => {
```

Comment:
Why are the code additions not optional? What if the library user does not require telemetry collection and opts out?

Reply:
The general design guidance for libraries with open telemetry is to implement the tracing. If the library user sets collectors and exporters, it'll go where asked to go. If they don't, all of that just pipes to null. This way libraries don't have to implement settings which could be hard to control with nested of libs about enabling or not tracing.